invenio_rdm_records/resources/serializers/marcxml/schema.py

```diff
@@ -255,4 +255,9 @@ def keys_to_tags(self, data, many, **kwargs):
             if key in changes:
                 data[changes[key][0]] = {changes[key][1]: data.pop(key)}
 
+        doi = data.get("doi", None)
```

Comment:
you can merge line 258 and 260. If you pass a default value to `pop` it will not throw an exception.

```python
>>> a = {"test": 1}
>>> a.pop("no", None)
>>> a.pop("test")
1
```

so you could do something like:

```python
doi = data.pop("doi", None)
if doi:
    data[...] = ...
```

Reply:
Oh didn't know that was a thing thx!